plutus-tx-plugin/test/IntegerLiterals/NoStrict/9.6/integerLiterals-NoStrict.pir.golden

```diff
@@ -0,0 +1,76 @@
+(program
+  1.1.0
+  (let
+    (nonrec)
+    (termbind
+      (strict)
+      (vardecl integerNegate (fun (con integer) (con integer)))
```

Comment:
I'd like to see some examples in tests where we have some optimization. I hope we inline this: it's a guarantee win if the argument is a constant since we'll then reduce it.

Reply:
It is now inlined, thanks to `annAlwaysInline`!

Reply:
I'm not sure if that's the right call either. It _might_ be bad to inline if the argument isn't a constant. Probably we should just try and make sure the inliner is smart enough to see that. Not sure.

Reply:
There's no other good way to make it inline, because `let !x = negateInteger 100` is not considered safe to inline, unless we somehow special-case it in the inliner. And the argument is almost always constant.

Reply:
> There's no other good way to make it inline, because let !x = negateInteger 100 is not considered safe to inline

That's different. I'm talking about inlining the `negateInteger` call, not `x`. And `negateInteger` is certainly safe to inline, it just won't tend to be smaller, which is why we won't inline it.

>  And the argument is almost always constant.

... in your test cases. Now the plugin supports it, people can just write `-(some expression)` throughout their code. Who knows what's in there. It's probably not a big deal, but worth discussing.